packages/modules/src/modules/banners/common/choiceCard/ChoiceCards.tsx

```diff
@@ -1,40 +1,27 @@
 import React, { useEffect } from 'react';
-import { ChoiceCardGroup } from '@guardian/src-choice-card';
 import { css, SerializedStyles } from '@emotion/react';
 import { from } from '@guardian/src-foundations/mq';
 import { HasBeenSeen, useHasBeenSeen } from '../../../../hooks/useHasBeenSeen';
-import { ChoiceCardAmountButtons } from './ChoiceCardAmountButtons';
-import { ChoiceCardFrequencyTabs } from './ChoiceCardFrequencyTabs';
+import { ChoiceCardInteractive } from './ChoiceCardInteractive';
 import { SupportCta } from './SupportCta';
 import { PaymentCards } from '../PaymentCards';
 import { BannerTextContent } from '../../common/types';
-import { ContributionFrequency } from '@sdc/shared/src/types';
-import {
-    OphanComponentEvent,
-    AmountsCardData,
-    ContributionType,
-    Tracking,
-} from '@sdc/shared/src/types';
+import { OphanComponentEvent, SelectedAmountsVariant, Tracking } from '@sdc/shared/src/types';
 import type { ReactComponent } from '../../../../types';
-import { ChoiceCardSettings } from '../../momentTemplate/settings';
+import { ChoiceCardSelection } from '../../../shared/helpers/choiceCards';
 
-export type ChoiceCardBannerComponentId = 'choice-cards-buttons-banner-blue';
-
-export interface ChoiceCardSelection {
-    frequency: ContributionFrequency;
-    amount: number | 'other';
+export interface ChoiceCardSettings {
+    buttonColour?: string;
 }
 
 interface ChoiceCardProps {
     selection?: ChoiceCardSelection;
     setSelectionsCallback: (choiceCardSelection: ChoiceCardSelection) => void;
     submitComponentEvent?: (event: OphanComponentEvent) => void;
     currencySymbol: string;
-    componentId: ChoiceCardBannerComponentId;
+    componentId: string;
     getCtaText: (contentType: 'mainContent' | 'mobileContent') => string;
-    amounts?: AmountsCardData;
-    amountsTestName?: string;
-    amountsVariantName?: string;
+    amountsTest: SelectedAmountsVariant;
```

Comment:
In the places where this component is used, this data is potentially undefined according to the types.
So should this field be optional?
`amountsTest?: SelectedAmountsVariant;`
Or fix it further up. It's a bit worrying that it builds with these type errors!

Reply:
I guess this all stems from the fact that `choiceCardAmounts` is optional in the banner props model -
https://github.com/guardian/support-dotcom-components/blob/main/packages/shared/src/types/props/banner.ts#L56

Reply:
Good catch! I think we should keep it as optional. We test to see if the amountsTest object has been supplied and bale out of the component if not. Updated code pushed.

Reply:
Thanks. At some point I'll try to understand why typescript wasn't picking this up - intellij spotted it!